integration/setup/config.go

```diff
@@ -0,0 +1,230 @@
+//go:build k8srequired
+// +build k8srequired
+
+package setup
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/apptest"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/k8sclient/v6/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/app-admission-controller/integration/env"
+	"github.com/giantswarm/app-admission-controller/integration/helpers"
+)
+
+type TestConfig struct {
+	AppTest    apptest.Interface
+	K8sClients k8sclient.Interface
+	Logger     micrologger.Logger
+}
+
+func NewConfig() (TestConfig, error) {
+	var err error
+
+	var logger micrologger.Logger
+	{
+		c := micrologger.Config{}
+
+		logger, err = micrologger.New(c)
+		if err != nil {
+			return TestConfig{}, microerror.Mask(err)
+		}
+	}
+
+	var k8sClients *k8sclient.Clients
+	{
+		c := k8sclient.ClientsConfig{
+			Logger: logger,
+
+			KubeConfigPath: env.KubeConfig(),
+		}
+
+		k8sClients, err = k8sclient.NewClients(c)
+		if err != nil {
+			return TestConfig{}, microerror.Mask(err)
+		}
+	}
+
+	var appTest apptest.Interface
+	{
+		c := apptest.Config{
+			Logger: logger,
+
+			KubeConfigPath: env.KubeConfig(),
+		}
+
+		appTest, err = apptest.New(c)
+		if err != nil {
+			return TestConfig{}, microerror.Mask(err)
+		}
+	}
+
+	c := TestConfig{
+		AppTest:    appTest,
+		K8sClients: k8sClients,
+		Logger:     logger,
+	}
+
+	return c, nil
+}
+
+func (tc *TestConfig) CreateNamespace(ctx context.Context, name string) error {
```

Comment:
This is nicely structured and I like how you reuse the ensureExecuted method.

Just one nit. Can you order the methods alphabetically?

Reply:
sure! It's done